### PR TITLE
♻️ Refactors service listing queries and 🗃️ drops ``service_latest`` cache table

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

This PR address some refactoring agreed with some the reviewers after  https://github.com/ITISFoundation/osparc-simcore/pull/4038 . Specifically:

- 🗃️ removes ``services_latest`` table from pg
    -  ♻️ removes step to fill ``services_latest`` in periodic background task of ``catalog`` service
- ✨ This table is replaced by a subquery  in ``simcore_postgres_database.utils_services``

## Related issue/s

- follows from https://github.com/ITISFoundation/osparc-simcore/pull/4038 
- related to ITISFoundation/osparc-issues#683
- closes https://github.com/ITISFoundation/osparc-simcore/issues/4032

## How to test

- test coverage
- **Manual test**:
   - deploy stack and open web api swagger site:
   - ``GET /v0/services`` should list latest versions of (see ``view_url`` to check version)

### CODE REVIEW

File: services/web/server/src/simcore_service_webserver/studies_dispatcher/_catalog.py [hunk 1]
```diff
@@ -62,6 +62,9 @@ async def iter_latest_osparc_services(
     engine: Engine = get_database_engine(app)
     settings: StudiesDispatcherSettings = get_plugin_settings(app)
 
+    # Select query for latest version of the service
+    latest_view = create_select_latest_services_query().alias("latest_view")
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/studies_dispatcher/_catalog.py:66]: I would stick with the previous name, this way we do not confuse people. A view is not a query in SQL language
```suggestion
    latest_services = create_select_latest_services_query().alias("latest_view")
```

**Review by GitHK — APPROVED** [review]
Vey nice 👍

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

**Review by sanderegg — COMMENTED** [review]
great! thank you for the change!

**Review by sanderegg — APPROVED** [review]

**Comment by pcrespov** [comment]
- Manually tested in staging! :heavy_check_mark:

**Pull request merged**